AppCenter/AppCenterTests/MSAppCenterTests.m

```diff
@@ -975,4 +975,106 @@ - (void)testNoUserIdWhenSetUserIdIsNotCalledInNextVersion {
   XCTAssertNil([[MSUserIdContext sharedInstance] userIdAt:[NSDate dateWithTimeIntervalSince1970:5000]]);
 }
 
+- (void)testServiceNotificationNoRegisteredDelegate {
+  
+  // If
+  // Notification data.
+  NSDictionary<NSString *, NSString *> *data = @{@"key" : @"value"};
+
+  // When
+  [self.sut forwardServiceNotification:data];
+
+  // Just checking this doesn't crash.
+}
+
+- (void)testServiceNotificationAddDelegate {
+
+  // If
+  // Creating delegates.
+  id delegate1 = OCMProtocolMock(@protocol(MSServiceNotificationDelegate));
+  id delegate2 = OCMProtocolMock(@protocol(MSServiceNotificationDelegate));
+
+  // Registering delegates.
+  [self.sut addServiceNotificationDelegate:delegate1];
+  [self.sut addServiceNotificationDelegate:delegate2];
+
+  // Notification data.
+  NSDictionary<NSString *, NSString *> *data = @{@"key" : @"value"};
+
+  // When
+  [self.sut forwardServiceNotification:data];
+
+  // Then
+  OCMVerify([delegate1 appCenter:_sut didReceiveServiceNotification:data]);
```

Comment:
We usually postfix mocks names (it's not consistent everywhere though). But that would make `delegate1` and `delegate2` renamed to `delegate1Mock` `delegate2Mock`